EIPS/eip-7748.md

```diff
@@ -0,0 +1,292 @@
+author: Guillaume Ballet (@gballet), Ignacio Hagopian (@jsign), Gajinder Singh (@g11tech), Ansgar Dietrichs (@adietrichs), Gottfried Herold (@GottfriedHerold), Jamie Lokier (@jlokier), Tanishq Jasoria (@tanishqjasoria), Parithosh Jayanthi (@parithosh), Gabriel Rocheleau (@gabrocheleau), Karim Taam (@matkt)
+discussions-to: https://ethereum-magicians.org/t/eip-7748-state-conversion-to-verkle-tree/20625
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-07-23
+requires: 7612
+---
+
+## Abstract
+
+This EIP proposes a procedure to convert, on each block, a fixed number of key-values from the existing  Merkle Patricia Tree (MPT) to the Verkle Tree (VKT).
+
+## Motivation
+
+The accounts state is too large to wait for transactions to organically move all of them to the VKT through the Overlay Tree. Thus, we need a strategy to convert all the state within a reasonable time. The state conversion completion allows removing the Overlay Tree abstraction introduced in [EIP-7612](./eip-7612.md) and use directly the VKT for all state access.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Constants
+
+| Parameter                    | value | Description                                                    |
+| ---------------------------- | ----- | -------------------------------------------------------------- |
+| `CONVERSION_START_TIMESTAMP` | `TBD` | Timestamp at which the conversion starts.                      |
+| `CONVERSION_STRIDE`          | `TBD` | Maximum number of _conversion units_ to be converted per block |
+
+A _conversion unit_ is:
+
+- A contract storage slot.
+- A contract code. (i.e. all the code is a single _conversion unit_)
+- An account data. (e.g. balance, nonce, code-hash)
+
+### Changes to the execution spec
+
+Include the following code in the existing `apply_body(...)` function:
+
+```python
+def apply_body(state: State, ...) -> Tuple[Uint, Root, Root, Bloom, State, Root]:
+    ...
+    # <new_code>
+    if block_time >= CONVERSION_START_TIMESTAMP and not state._conversion_finished:
+        block_state_conversion(state, CONVERSION_STRIDE)
+    # </new_code>
+    
+    for i, tx in enumerate(map(decode_transaction, transactions)):
+        ...
+    ...
+```
+
+Before executing txs, it calls `block_state_conversion(...)` (described below) which performs a state conversion step for this block.
+
+In `state.py`, add the following code:
+
+```python
+@dataclass
+class StoragePhase:
+    """
+    The account next conversion step continues converting the 
+    storage-slot with key greater or equal next_key. 
+    If there isn't such storage-slot, the account must move to
+    BasicDataPhase.
+    """
+    next_key: Bytes   
+   
+@dataclass
+class BasicDataPhase:
```

Comment:
The BasicData naming is a bit confusing to me because there is also a field named BasicData in the verkle leaf. However here, if I'm following, BasicData includes not only BasicData from the verkle leaf (version,balance,nonce,codesize) but also codehash and contract code.

Reply:
Yes, you're right. Probably `BasicData` is falling short if we try to keep the semantics clean and only have a single meaning (i.e: the VKT one).

Probably it will be better to choose another name. Open to suggestions!

Reply:
Let's continue in https://github.com/ethereum/EIPs/pull/8780!